collector/mongos/collections_status.go

```diff
@@ -98,70 +100,58 @@ func (collStatList *CollectionStatList) Describe(ch chan<- *prometheus.Desc) {
 	collectionIndexesSize.Describe(ch)
 }
 
-var (
-	logSuppressCS = make(map[string]bool)
-)
+var logSuppressCS = shared.NewSyncStringSet()
 
-// GetCollectionStatList returns stats for a given database
+const keyCS = ""
+
+// GetCollectionStatList returns stats for all non-system collections.
 func GetCollectionStatList(client *mongo.Client) *CollectionStatList {
 	collectionStatList := &CollectionStatList{}
 	dbNames, err := client.ListDatabaseNames(context.TODO(), bson.M{})
 	if err != nil {
-		_, logSFound := logSuppressCS[""]
-		if !logSFound {
-			log.Errorf("%s. Collection stats will not be collected. This log message will be suppressed from now.", err)
-			logSuppressCS[""] = true
+		if !logSuppressCS.Contains(keyCS) {
+			log.Warnf("%s. Collection stats will not be collected. This log message will be suppressed from now.", err)
+			logSuppressCS.Add(keyCS)
 		}
 		return nil
 	}
-	delete(logSuppressCS, "")
+
+	logSuppressCS.Delete(keyCS)
 	for _, dbName := range dbNames {
-		c, err := client.Database(dbName).ListCollections(context.TODO(), bson.M{}, options.ListCollections().SetNameOnly(true))
+		if common.IsSystemDB(dbName) {
+			continue
+		}
+
+		collNames, err := client.Database(dbName).ListCollectionNames(context.TODO(), bson.M{})
 		if err != nil {
-			_, logSFound := logSuppressCS[dbName]
-			if !logSFound {
-				log.Errorf("%s. Collection stats will not be collected for this db. This log message will be suppressed from now.", err)
-				logSuppressCS[dbName] = true
+			if !logSuppressCS.Contains(dbName) {
+				log.Warnf("%s. Collection stats will not be collected for this db. This log message will be suppressed from now.", err)
+				logSuppressCS.Add(dbName)
 			}
-		} else {
+			continue
+		}
 
-			type collListItem struct {
-				Name string `bson:"name,omitempty"`
-				Type string `bson:"type,omitempty"`
+		logSuppressCS.Delete(dbName)
+		for _, collName := range collNames {
+			if common.IsSystemCollection(collName) {
```

Comment:
Is it possible that https://github.com/percona/mongodb_exporter/pull/177/files#diff-007e82e9748a45f49784e15a4c20541cR125 returns collection from system database?

Reply:
Yes, `ListCollectionNames` returns all collections names in current db, even system. Error raised when you try to get access to such system collection.

Reply:
I mean mongo db there is system collections in non-system databases, `system.profile` for example.

Reply:
System databases will be skipped at line https://github.com/percona/mongodb_exporter/pull/177/files#diff-007e82e9748a45f49784e15a4c20541cR121